torch/csrc/utils/tensor_new.cpp

```diff
@@ -771,16 +773,53 @@ void _validate_sparse_coo_tensor_args(c10::DispatchKey dispatch_key, at::ScalarT
   at::native::_validate_sparse_coo_tensor_args(indices, values, r.intlist(2));
 }
 
+
+
+
+
 Tensor tensor_ctor(c10::DispatchKey dispatch_key, at::ScalarType scalar_type, PyObject* args, PyObject* kwargs) {
+
   static PythonArgParser parser({
     "tensor(PyObject* data, *, ScalarType dtype=None, Device? device=None, bool pin_memory=False, bool requires_grad=False, DimnameList? names=None)",
   });
 
   constexpr int ctor_num_args = 6;
   ParsedArgs<ctor_num_args> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
+
   if (r.idx == 0) {
     PyObject* data = r.pyobject(0);
+
+    PyObject* meta_str = PyUnicode_DecodeFSDefault("meta");
+    PyObject* ellipsis_str = PyUnicode_DecodeFSDefault("Ellipsis");
+
+    PyObject* data_passed = PyObject_Str(data);
+    PyObject* device_passed = r.pyobject(2);
+
+    if(PyObject_RichCompareBool(device_passed, meta_str, Py_EQ)==1 && PyObject_RichCompareBool(data_passed, ellipsis_str, Py_EQ)==1){
+
+        c10::TensorOptions options = typeIdWithDefault(r, 2, dispatch_key);
+        at::ScalarType scalar_type = r.scalartypeWithDefault(1, scalar_type);
+        c10::optional<Device> device_opt;
+        bool copy_variables = true;
+        bool copy_numpy = true;
+        bool type_inference = r.isNone(1);
+        bool pin_memory = r.toBool(3);
+        auto sizes = compute_sizes(data);
+        Tensor new_tensor;
+        at::AutoDispatchBelowADInplaceOrView guard;  // TODO: remove
+        at::tracer::impl::NoTracerDispatchMode tracer_guard;
+        new_tensor = at::empty(0);
+        bool args_requires_grad = r.toBool(4);
+        new_tensor.detach_(); // ensure new_tensor a leaf node
+        new_tensor.set_requires_grad(args_requires_grad);
```

Comment:
You need to respect sizes. `tensor(..., device='meta', size=(2,))` should work too.